lightning/src/util/config.rs

```diff
@@ -269,7 +274,7 @@ impl_writeable_tlv_based!(ChannelConfig, {
 	(2, cltv_expiry_delta, required),
 	(3, force_close_avoidance_max_fee_satoshis, (default_value, 1000)),
 	(4, announced_channel, required),
-	(6, commit_upfront_shutdown_pubkey, required),
+	(6, commit_upfront_shutdown_pubkey, option),
```

Comment:
Actually, this presents a backwards-compatibility concern - one thing we explicitly support in LDK is for users to downgrade (at least a version or two) and still be able to open objects serialized with the newest version. Here, if a user did that, they'd fail to read the `ChannelConfig` because the value went from always present (and required) to defaulting to `None`. We can paper over this, though, by always defaulting to creating `commit_upfront_shutdown_pubkey` and setting it to `Some(...)` but then setting the "correct" value on the write side of channel to ensure old versions read the current/new value.